src/main/org/apache/tools/ant/taskdefs/optional/junitlauncher/JUnitLauncherTask.java

```diff
@@ -0,0 +1,508 @@
+            Thread.currentThread().setContextClassLoader(executionCL);
+            final Launcher launcher = LauncherFactory.create();
+            final List<TestRequest> requests = buildTestRequests();
+            for (final TestRequest testRequest : requests) {
+                try {
+                    final TestDefinition test = testRequest.getOwner();
+                    final LauncherDiscoveryRequest request = testRequest.getDiscoveryRequest().build();
+                    final List<TestExecutionListener> testExecutionListeners = new ArrayList<>();
+                    // a listener that we always put at the front of list of listeners
+                    // for this request.
+                    final Listener firstListener = new Listener();
+                    // we always enroll the summary generating listener, to the request, so that we
+                    // get to use some of the details of the summary for our further decision making
+                    testExecutionListeners.add(firstListener);
+                    testExecutionListeners.addAll(getListeners(testRequest, executionCL));
+                    final PrintStream originalSysOut = System.out;
+                    final PrintStream originalSysErr = System.err;
+                    try {
+                        firstListener.switchedSysOutHandle = trySwitchSysOut(testRequest);
+                        firstListener.switchedSysErrHandle = trySwitchSysErr(testRequest);
+                        launcher.execute(request, testExecutionListeners.toArray(new TestExecutionListener[testExecutionListeners.size()]));
+                    } finally {
+                        // switch back sysout/syserr to the original
+                        try {
+                            System.setOut(originalSysOut);
+                        } catch (Exception e) {
+                            // ignore
+                        }
+                        try {
+                            System.setErr(originalSysErr);
+                        } catch (Exception e) {
+                            // ignore
+                        }
+                    }
+                    handleTestExecutionCompletion(test, firstListener.getSummary());
+                } finally {
+                    try {
+                        testRequest.close();
+                    } catch (Exception e) {
+                        // log and move on
+                        log("Failed to cleanly close test request", e, Project.MSG_DEBUG);
+                    }
+                }
+            }
+        } finally {
+            Thread.currentThread().setContextClassLoader(previousClassLoader);
+        }
+    }
+
+    /**
+     * @return Creates and returns the a {@link Path} which will be used as the classpath of this
+     * task. This classpath will then be used for execution of the tests
+     */
+    public Path createClassPath() {
+        this.classPath = new Path(getProject());
```

Comment:
if users specify multiple nested classpath elements, only the last one will be used. You may want to return an already created `this.classpath` or move on to an `add` method and collect the paths inside of a wrapper `Path`.

Reply:
You are right. I hadn't taken that into account. I have updated the PR to change this part of the code.